modules/mtl/arrow-mtl-data/src/test/kotlin/arrow/mtl/ComposedInstancesTest.kt

```diff
@@ -87,26 +90,32 @@ class ComposedInstancesTest : UnitSpec() {
 
     val cf: (Int) -> Kind<Nested<ForOption, ForNonEmptyList>, Int> = { Some(it.nel()).nest() }
 
+    val G = Gen.intSmall().map(cf)
+
+    val GK = object : GenK<Nested<ForOption, ForNonEmptyList>> {
+      override fun <A> genK(gen: Gen<A>): Gen<Kind<Nested<ForOption, ForNonEmptyList>, A>> = gen.map { Some(it.nel()).nest() }
+    }
```

Comment:
You can actually get a good generator for all cases by using something like `gen.nonEmptyList().option()` to generate every case possible

Reply:
The same goes for every gen/cf function here, and also eqK instances.

Reply:
fixed the generator and tried to get rid of cf functions.

i struggled here to find a GenK replacement for cf2. maybe you could help here

```
val cf2: (Int) -> Kind<Nested<ForOption, Conested<ForFunction1, Int>>, Int> = { x: Int ->
      Some({ y: Int -> x + y }.k<Int, Int>().conest()).nest()
    }
```

Reply:
Coming up with `GenK` for nested stuff is weird. I'll take a look later today/tomorrow. Till then just ignore it.

Reply:
after doing the Sum EqK i revisited this. thin i got it now :-)